pkg/metrics/metrics.go

```diff
@@ -31,6 +31,11 @@ const (
 	// ErrorProxy is the value used to notify errors on Proxy.
 	ErrorProxy = "proxy"
 
+	// HistogramFactor configures the factor used for exponential/native histogram in Prometheus client
+	HistogramFactor = 1.1
+	// HistogramMaxBuckets configures the maximum number of buckets that will be exported in Prometheus client
+	HistogramMaxBuckets = 128
```

Comment:
With this new feature, you could endup with unbounded dynamic buckets, since each observation may create a new bucket in the histogram. To avoid problems with memory, we introduce this parameter to limit the amount of buckets exported by the client golang.

Reply:
Also documented here: https://github.com/prometheus/client_golang/blob/main/prometheus/histogram.go#L442-L466